rhizome/host/lib/cloud_hypervisor.rb

```diff
@@ -13,6 +14,34 @@ def path
     end
   }
 
+  FirmwareClass = Struct.new(:arch, :version) {
+    def url
+      "https://github.com/ubicloud/build-edk2-firmware/releases/download/edk2-stable#{version}-#{arch}/CLOUDHV-#{arch}.fd"
+    end
+
+    def firmware_root
+      "/opt/fw"
+    end
+
+    def path
+      "#{firmware_root}/CLOUDHV-#{arch}-#{version}.fd"
+    end
+
+    def download
+      return if File.exist?(path)
+      FileUtils.mkdir_p(firmware_root)
+      temp_path = path + ".tmp"
+      curl(temp_path, url)
+      FileUtils.mv(temp_path, path)
```

Comment:
I suppose we should flush this (see safe_write_to_file, though it may need a variant that accepts a block to let a subprocess do the writing.

I suppose also given the size, it'd be nice (but not *necessary* if too bloated) to digest the file similar to recent changes to images, now that the standards of adequacy are going up.

Reply:
Extended `safe_write_to_file` and using it to produce the firmware file. Also, it is now digested.